pkg/ruler/ruleset.go

```diff
@@ -78,7 +78,7 @@ func NewRuleset(logger *zap.SugaredLogger) *Ruleset {
 	runAsNonRootRule := Rule{
 		Predicate: rules.RunAsNonRoot,
 		ID:        "RunAsNonRoot",
-		Selector:  "containers[] .securityContext .runAsNonRoot == true",
+		Selector:  ".spec, .spec.containers[] | .securityContext .runAsNonRoot == true",
```

Comment:
```suggestion
		Selector:  "(.spec | .spec.containers[]) .securityContext .runAsNonRoot == true",
```

Is it not more like so?

Reply:
The selector is used to get both `.spec` and `.spec.containers[]` in order to get both the security context for the pod and the containers.

Considering the following JSON:

```json
{
  "apiVersion": "v1",
  "kind": "Pod",
  "metadata": {
    "name": "poc"
  },
  "spec": {
    "securityContext": {
      "runAsNonRoot": true
    },
    "containers": [
      {
        "name": "name",
        "image": "image",
        "securityContext": {
          "runAsNonRoot": false
        }
      },
      {
        "name": "name2",
        "image": "image2",
        "securityContext": {
          "runAsNonRoot": false
        }
      }
    ]
  }
}
````

Using JQ, we get the `runAsNonRoot` from the all the security contexts.

```
$ jq '.spec, .spec.containers[] | .securityContext.runAsNonRoot' pod.json
true
false
true
```